packaging_automation/common_tool_methods.py

```diff
@@ -171,7 +184,12 @@ def is_merge_commit(commit: Commit):
 @validate_parameters
 def get_version_details(version: is_version(str)) -> Dict[str, str]:
     version_parts = version.split(".")
-    return {"major": version_parts[0], "minor": version_parts[1], "patch": version_parts[2]}
+    release_stage = 'stable'
+    if len(version_parts) == 3:
+        if '_' in version:
+            stage_parts = version_parts[2].split('_')
+            release_stage = stage_parts[1]
```

Comment:
Maybe put some comments and consider using release_type instead of release_stage:
Also, can it ever be the case that `len(version_parts)` is not equal to 3 ?

```suggestion
    release_type = 'stable'
    if '_' in version:
        # Having '_' in version string means that the last part of
        # it explicitly specifies release type.
        patch_version_parts = version_parts[2].split('_')
        release_type = patch_version_parts[1]
```

Reply:
Used a method instead of the explicit line. This method name explains better I think